compiler/passes/insertWideReferences.cpp

```diff
@@ -1489,6 +1491,11 @@ static void narrowWideClassesThroughCalls()
             stmt->insertBefore(new CallExpr(PRIM_MOVE, var, sym->copy()));
           }
           else if (narrowType.isRef() || narrowType.type()->symbol->hasFlag(FLAG_DATA_CLASS)) {
+            // Insert a local check because we cannot pass narrow references to
+            // remote data to external routines
+            if (!fNoLocalChecks)
+              stmt->insertBefore(new CallExpr(PRIM_LOCAL_CHECK, sym->copy(), buildCStringLiteral(astr("references to remote data cannot be passed to external routines like '", fn->name, "'"))));
```

Comment:
Technically, there might be non-extern functions with FLAG_LOCAL_ARGS, so it might be useful to make different error messages depending on whether the function is marked extern. In particular `export` functions also have this flag. You could also reword it to refer to "functions accepting only local arguments".

Reply:
Oh, thanks for pointing that out.  Note that it actually should be fine for Chapel to pass wide things to exported functions—the requirement/expectation that the incoming arguments are narrow only applies to calls that are actually coming from outside of Chapel.  I'd guess that this probably means we have problems today if we called both using a wide reference from within Chapel and a narrow one from outside of Chapel, but in any case, you've pointed out that my insertion of the check is incorrect for exported functions.

Reply:
I've opened PR #11562 to add a future showing that I don't think export functions should assume all their arguments are local for calls that originate from within Chapel.